test/framework/src/main/java/org/elasticsearch/transport/MockTcpTransport.java

```diff
@@ -294,26 +333,23 @@ public void onFailure(Exception e) {
                 @Override
                 protected void doRun() throws Exception {
                     StreamInput input = new InputStreamStreamInput(new BufferedInputStream(activeChannel.getInputStream()));
+                    // as per interruption policy of CancellableThreads we don't need to check for interrupts.
```

Comment:
I think there is still a extremely small chance someone interrupts the thread before cancellable threads is called and since we clean up any interrupt flag over before we start, it won't block the read. So I think what you had is strictly speaking needed. However, for the testing infra part, I don't think we're required to honor that interrupt and we'll be killed anyway. I'm OK with the current code, but the comment is not correct.